dev-libs/libclc/libclc-0.2.0_pre20180610.ebuild

```diff
@@ -0,0 +1,85 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+PYTHON_COMPAT=( python2_7 python3_4 python3_5 python3_6 )
+
+EGIT_REPO_URI="http://llvm.org/git/${PN}.git
+	https://github.com/llvm-mirror/${PN}.git"
+EGIT_COMMIT="96627043299f52bf08f6a68aa6d00b8037a0a2b0"
+
+if [[ ${PV} = 9999* ]]; then
+	GIT_ECLASS="git-r3"
+	EXPERIMENTAL="true"
+else
+	GIT_ECLASS="vcs-snapshot"
+fi
+
+inherit llvm prefix python-any-r1 toolchain-funcs ${GIT_ECLASS}
+
+DESCRIPTION="OpenCL C library"
+HOMEPAGE="http://libclc.llvm.org/"
+
+if [[ ${PV} = 9999* ]]; then
+	SRC_URI="${SRC_PATCHES}"
+else
+	SRC_URI="https://github.com/llvm-mirror/libclc/archive/${EGIT_COMMIT}.tar.gz -> ${P}.tar.gz
+		${SRC_PATCHES}"
+fi
+
+LICENSE="|| ( MIT BSD )"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
+IUSE_VIDEO_CARDS="video_cards_nvidia video_cards_r600 video_cards_radeonsi"
+IUSE="${IUSE_VIDEO_CARDS}"
+REQUIRED_USE="
+	|| ( video_cards_nvidia video_cards_r600 video_cards_radeonsi )
+"
+
+RDEPEND="
+	|| (
+		sys-devel/clang:7
+		sys-devel/clang:6
+		sys-devel/clang:5
+		sys-devel/clang:4
+		>=sys-devel/clang-3.9:0
+	)"
+DEPEND="${RDEPEND}
+	${PYTHON_DEPS}"
+
+LLVM_MAX_SLOT=7
+
+llvm_check_deps() {
+	has_version "sys-devel/clang:${LLVM_SLOT}"
+}
+
+src_prepare() {
+	default
+	if use prefix; then
+		hprefixify configure.py
+	fi
+}
+
+pkg_setup() {
+	# we do not need llvm_pkg_setup
+	python-any-r1_pkg_setup
+}
+
+src_configure() {
+	LIBCLC_TARGETS=""
+
+	use video_cards_nvidia && LIBCLC_TARGETS="${LIBCLC_TARGETS} nvptx-- nvptx64-- nvptx--nvidiacl nvptx64--nvidiacl"
+	use video_cards_r600 && LIBCLC_TARGETS="${LIBCLC_TARGETS} r600--"
+	use video_cards_radeonsi && LIBCLC_TARGETS="${LIBCLC_TARGETS} amdgcn-- amdgcn-mesa-mesa3d amdgcn--amdhsa"
+
+	[ "x${LIBCLC_TARGETS}" != "x" ] || die
```

Comment:
Don't do the ugly x-stuff in ebuilds, use full bash `[[ ... ]]`.

Reply:
I kept single [ and changed this to count the array elements